### PR TITLE
B2BDS-205 Disabling individual Checkboxes

### PR DESCRIPTION
**Pull request by AnnaHoege**

### CODE REVIEW

**Review by BalajiVijayakumar1195 — APPROVED** [review]

File: packages/core-components/src/components/checkbox-group/checkbox-group.tsx [hunk 1]
```diff
@@ -74,8 +78,8 @@ export class CheckboxGroupComponent {
         node.disabled = true;
       });
     } else {
-      nodes.forEach(node => {
-        node.disabled = false;
+      nodes.forEach((node, index) => {
```
Comment (BalajiVijayakumar1195) [inline packages/core-components/src/components/checkbox-group/checkbox-group.tsx:81]: Won't it be a problem to rely on index ?

Comment (BalajiVijayakumar1195) [reply packages/core-components/src/components/checkbox-group/checkbox-group.tsx:81]: For instance,
```
<g>
<c id="1">
<c id="2" disabled>
<g>
```

Disable group
```
<g disable>
<c id="1" disabled>
<c id="2" disabled>
<g>
```

Based on some condition render UI
```
<g disable>
<c id="2" disable>
<c id="1" disabled>
<g>
```

Enable group
```
<g>
<c id="2">
<c id="1" disabled>
<g>
```

File: packages/core-components/src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -44,6 +44,9 @@ export class CheckboxComponent {
   /** Whether or not the checkbox is rendered with error styles. Defaults to false. */
   @Prop() invalid?: boolean = false;
 
+  /** @internal whether the parent checkbox-group is disabled. */
+  @Prop() groupDisabled = false;
```
Comment (BalajiVijayakumar1195) [inline packages/core-components/src/components/checkbox/checkbox.tsx:48]: Should be optional right ?

**Review by BalajiVijayakumar1195 — APPROVED** [review]

File: packages/core-components/src/components/checkbox/checkbox.tsx (hunk 2 above)
Comment (AnnaHoege) [reply packages/core-components/src/components/checkbox/checkbox.tsx:48]: It isn't reflected and internal and by setting a default we don't run the risk of it being undefined per default.

**Pull request merged**